src/services/roleReactor.ts

```diff
@@ -1,36 +1,30 @@
 import type { GuildMember, MessageReaction } from "discord.js";
 import type { GuildMessage } from "../index.types";
 
-const enum ReactorRoles {
-  MPLUSDPS = "⚔",
-  MPLUSTANK = "🛡",
-  MPLUSHEALS = "🚑",
-  BGDPS = "🔥",
-  BGHEALS = "❤",
-  ARENADPS = "🥇",
-  ARENAHEALS = "🥈",
+export interface ReactorRole {
+  emoji: string;
+  data: {
+    name: string;
+    color?: string;
+  };
 }
 
-const roleList: readonly [ReactorRoles, string][] = [
-  [ReactorRoles.MPLUSDPS, "M+ DPS"],
-  [ReactorRoles.MPLUSTANK, "M+ Tank"],
-  [ReactorRoles.MPLUSHEALS, "M+ Healer"],
-  [ReactorRoles.BGDPS, "BG DPS"],
-  [ReactorRoles.BGHEALS, "BG Healer"],
-  [ReactorRoles.ARENADPS, "Arena DPS"],
-  [ReactorRoles.ARENAHEALS, "Arena Healer"],
-];
-
-const extractRoleFromEmoji = (reaction: MessageReaction): string | undefined => {
-  const emoji = reaction.emoji.toString();
-  return roleList.find(([roleEmoji]) => roleEmoji === emoji)?.[1];
-};
-
 export class RoleReactor {
+  private roleList: ReactorRole[];
   private cachedReactors = new Set<string>();
+
+  constructor(roles: ReactorRole[]) {
```

Comment:
Can just define the roleList in the constructor parameters:

```ts
constructor(private roleList: ReactorRole[]) {}
```

So to remove the boilerplate of the definition, then initialisation.

Reply:
Oh, and maybe mark it as `readonly` too if we never intend on mutating that list too